src/SignalRServiceExtension/Bindings/SignalRInputBindings/SignalRConnectionInputBinding.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    internal class SignalRConnectionInputBinding : IBinding
+    {
+        private const string HttpRequestName = "$request";
+        private readonly SignalRConnectionInfoAttribute attribute;
+        private readonly IAccessTokenProvider accessTokenProvider;
+        private readonly AzureSignalRClient azureSignalRClient;
+        private readonly ISignalRConnectionInfoConfigurer signalRConnectionInfoConfigurer;
+
+        public bool FromAttribute => true;
+
+        public SignalRConnectionInputBinding(SignalRConnectionInfoAttribute attribute, AzureSignalRClient azureSignalRClient, IAccessTokenProvider accessTokenProvider, ISignalRConnectionInfoConfigurer signalRConnectionInfoConfigurer)
+        {
+            this.accessTokenProvider = accessTokenProvider;
+            this.azureSignalRClient = azureSignalRClient;
+            this.attribute = attribute;
+            this.signalRConnectionInfoConfigurer = signalRConnectionInfoConfigurer;
+        }
+
+        public Task<IValueProvider> BindAsync(object value, ValueBindingContext context)
+        {
+            var bindingData = ((BindingContext)value).BindingData;
+
+            if (!bindingData.ContainsKey(HttpRequestName) || accessTokenProvider == null)
+            {
+                var info = azureSignalRClient.GetClientConnectionInfo(attribute.UserId, attribute.IdToken, attribute.ClaimTypeList);
+                return Task.FromResult((IValueProvider)new SignalRValueProvider(info));
+            }
+
+            var request = bindingData[HttpRequestName] as HttpRequest;
+
+            var tokenResult = accessTokenProvider.ValidateToken(request);
+
+            if (tokenResult.Status != AccessTokenStatus.Valid)
+            {
+                return Task.FromResult((IValueProvider)new SignalRValueProvider(null));
+            }
```

Comment:
the access token validation logic is same as that in `AccessTokenValidationInputBinding`, could be merged some how?

Reply:
Offline synced and understand the difference. please ignore this comment.

---
In reply to: [381091929](https://github.com/Azure/azure-functions-signalrservice-extension/pull/90#discussion_r381091929) [](ancestors = 381091929)